olive/model/handler/base.py

```diff
@@ -0,0 +1,66 @@
+import logging
+from abc import ABC, abstractmethod
+from typing import Any, ClassVar, Dict, List, Optional, Union
+
+from olive.constants import Framework, ModelFileFormat
+from olive.hardware.accelerator import Device
+from olive.model.mixin import CompositeMixin, IoConfigMixin, JsonMixin, ResourceMixin
+from olive.resource_path import OLIVE_RESOURCE_ANNOTATIONS
+
+logger = logging.getLogger(__name__)
+
+
+class OliveModelHandler(ABC, ResourceMixin, IoConfigMixin, JsonMixin, CompositeMixin):
+    """Abstraction for logical "Model", it contains model path and related metadata.
+
+    Each technique accepts Model as input, return Model as output.
+    """
+
+    model_type: Optional[str] = None
+    resource_keys: ClassVar[list] = ["model_path"]
+
+    def __init__(
+        self,
+        framework: Framework,
+        model_file_format: ModelFileFormat,
+        model_path: OLIVE_RESOURCE_ANNOTATIONS = None,
+        model_attributes: Optional[Dict[str, Any]] = None,
+    ):
+        self.framework = framework
```

Comment:
Could we add the comment to tell: 
if you add new Mixin into model handler, please make sure the member variable is initialized properly in model handler..